src/paulscode/android/mupen64plusae/profile/TouchscreenProfileActivity.java

```diff
@@ -385,37 +397,120 @@ private void hideSystemBars()
     @Override
     public boolean onTouch( View v, MotionEvent event )
     {
-        if( ( event.getAction() & MotionEvent.ACTION_MASK ) != MotionEvent.ACTION_DOWN )
-            return false;
-        
         int x = (int) event.getX();
         int y = (int) event.getY();
         
-        // Get the N64 index of the button that was pressed
-        int index = mTouchscreenMap.getButtonPress( x, y );
-        if( index != TouchMap.UNMAPPED )
+        if( ( event.getAction() & MotionEvent.ACTION_MASK ) == MotionEvent.ACTION_DOWN )
         {
-            String assetName = TouchMap.ASSET_NAMES.get( index );
-            String title = READABLE_NAMES.get( index );
+            initialX = x;
+            initialY = y;
+            dragIndex = TouchMap.UNMAPPED;
+            dragging = false;
             
-            // D-pad buttons are not holdable
-            if( DPAD.equals( assetName ) )
-                index = -1;
+            // ignore touches that start at the very bottom or top of the screen
+            View view = getWindow().getDecorView();
+            if (y < 10 || y > view.getHeight() - 10)
```

Comment:
Can this be improved? I just used 10 pixels because it needed some sort of dead zone for swipes, without being so large that it interferes with the ability to tap on a button to edit it.